imap-proto/src/parser/tests.rs

```diff
@@ -337,3 +337,33 @@ fn test_flags() {
         rsp => panic!("Unexpected response: {:?}", rsp),
     }
 }
+
+#[test]
+fn test_imap_body_structure() {
+    let mut test = br#"* 1569 FETCH (BODYSTRUCTURE (((("TEXT" "PLAIN" ("CHARSET" "ISO-8859-1") NIL NIL "QUOTED-PRINTABLE" 833 30 NIL NIL NIL)("TEXT" "HTML" ("CHARSET" "ISO-8859-1") NIL NIL "QUOTED-PRINTABLE" 3412 62 NIL ("INLINE" NIL) NIL) "ALTERNATIVE" ("BOUNDARY" "2__=fgrths") NIL NIL)("IMAGE" "GIF" ("NAME" "485039.gif") "<2__=lgkfjr>" NIL "BASE64" 64 NIL ("INLINE" ("FILENAME" "485039.gif")) NIL) "RELATED" ("BOUNDARY" "1__=fgrths") NIL NIL)("APPLICATION" "PDF" ("NAME" "title.pdf") "<1__=lgkfjr>" NIL "BASE64" 333980 NIL ("ATTACHMENT" ("FILENAME" "title.pdf")) NIL) "MIXED" ("BOUNDARY" "0__=fgrths") NIL NIL))"#.to_vec();
+    test.extend_from_slice(b"\r\n");
```

Comment:
```suggestion
    let test = br#"\
    * 1569 FETCH (\
        BODYSTRUCTURE (\
            (\
                (\
                    (\
                        "TEXT" "PLAIN" \
                        ("CHARSET" "ISO-8859-1") NIL NIL \
                        "QUOTED-PRINTABLE" 833 30 NIL NIL NIL\
                    )\
                    (\
                        "TEXT" "HTML" \
                        ("CHARSET" "ISO-8859-1") NIL NIL \
                        "QUOTED-PRINTABLE" 3412 62 NIL ("INLINE" NIL) NIL \
                    ) \
                    "ALTERNATIVE" ("BOUNDARY" "2__=fgrths") NIL NIL\
                )\
                (\
                    "IMAGE" "GIF" \
                    ("NAME" "485039.gif") "<2__=lgkfjr>" NIL \
                    "BASE64" 64 NIL ("INLINE" ("FILENAME" "485039.gif")) \
                    NIL\
                ) \
                "RELATED" ("BOUNDARY" "1__=fgrths") NIL NIL\
            )\
            (\
                "APPLICATION" "PDF" \
                ("NAME" "title.pdf") \
                "<1__=lgkfjr>" NIL "BASE64" 333980 NIL \
                ("ATTACHMENT" ("FILENAME" "title.pdf")) NIL\
            ) \
            "MIXED" ("BOUNDARY" "0__=fgrths") NIL NIL\
        )\
    )\r\n"#;
```

Like this?

Reply:
Yeah, that looks great!

Reply:
Then I believe all that needs to happen here to get things rolling again, is for @yageek to push the "Commit suggestion" button.